packer/builder/azure/arm/step_deploy_template.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+type StepDeployTemplate struct {
+	client *AzureClient
+	deploy func(resourceGroupName string, deploymentName string, templateParameters *TemplateParameters) error
+	say    func(message string)
+	error  func(e error)
+}
+
+func NewStepDeployTemplate(client *AzureClient, ui packer.Ui) *StepDeployTemplate {
+	var step = &StepDeployTemplate{
+		client: client,
+		say:    func(message string) { ui.Say(message) },
+		error:  func(e error) { ui.Error(e.Error()) },
+	}
+
+	step.deploy = step.deployTemplate
+	return step
+}
+
+func (s *StepDeployTemplate) deployTemplate(resourceGroupName string, deploymentName string, templateParameters *TemplateParameters) error {
+	factory := newDeploymentFactory(Linux)
+	deployment, err := factory.create(*templateParameters)
+	if err != nil {
+		return err
+	}
+
+	res, err := s.client.DeploymentsClient.CreateOrUpdate(resourceGroupName, deploymentName, *deployment)
+	if err != nil {
+		return err
+	}
+
+	s.client.DeploymentsClient.PollAsNeeded(res.Response.Response)
```

Comment:
Yes.

1st - ensure the call is accepted by Azure
2nd - ensure deployment reaches a completed state.  

You can ask for the 2nd until you're sure the 1st has completed.
I couldn't find an existing API to cover the 2nd, but maybe I'm missing something.

I've taken a TODO to double check these statements.